scripts/gke/use_gcr_for_all_images.sh

```diff
@@ -3,37 +3,70 @@
-#   docker push "${target}"
-# }
-# Example invocations:
-# sync_image prom/statsd-exporter:v0.6.0
-# sync_image quay.io/datawire/ambassador:0.37.0
-
-set -x
-
-if ks component list | awk '{print $1}' | grep -q "^argo$"; then
-  ks param set argo workflowControllerImage gcr.io/kubeflow-images-public/argoproj/workflow-controller:v2.2.0
-  ks param set argo uiImage gcr.io/kubeflow-images-public/argoproj/argoui:v2.2.0
-  ks param set argo executorImage gcr.io/kubeflow-images-public/argoproj/argoexec:v2.2.0
-fi
-
-if ks component list | awk '{print $1}' | grep -q "^cert-manager$"; then
-  ks param set cert-manager certManagerImage gcr.io/kubeflow-images-public/quay.io/jetstack/cert-manager-controller:v0.2.4
-  ks param set cert-manager certManagerIngressShimImage gcr.io/kubeflow-images-public/quay.io/jetstack/cert-manager-ingress-shim:v0.2.4
-fi
-
-if ks component list | awk '{print $1}' | grep -q "^ambassador$"; then
-  ks param set ambassador ambassadorImage gcr.io/kubeflow-images-public/quay.io/datawire/ambassador:0.37.0
-fi
-
-if ks component list | awk '{print $1}' | grep -q "^katib$"; then
-  ks param set katib modeldbDatabaseImage gcr.io/kubeflow-images-public/mongo:3.4
-  ks param set katib vizierDbImage gcr.io/kubeflow-images-public/mysql:8.0.3
-fi
+#
+# To sync the images to your registry use
+# PROJECT=$(PROJET) make copy-gcb
+
+set -xe
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+
+parseArgs() {
+  # Parse all command line options
+  while [[ $# -gt 0 ]]; do
+	# Parameters should be of the form
+	# --{name}=${value}
+	echo parsing "$1"
+	if [[ $1 =~ ^--(.*)=(.*)$ ]]; then
+	  name=${BASH_REMATCH[1]}
+	  value=${BASH_REMATCH[2]}
+
+	  eval ${name}="${value}"
+	elif [[ $1 =~ ^--(.*)$ ]]; then
+	name=${BASH_REMATCH[1]}
+	value=true
+	eval ${name}="${value}"
+	else
+	  echo "Argument $1 did not match the pattern --{name}={value} or --{name}"
+	fi
+	shift
+  done
+}
+
+usage() {
+  echo "Usage: use_gcr_for_all_images --registry=<REGISTRY>"
+}
+
+main() {
+	# List of required parameters
+  names=(registry)
+
+  missingParam=false
+  for i in ${names[@]}; do
+	if [ -z ${!i} ]; then
+	  echo "--${i} not set"
+	  missingParam=true
+	fi
+  done
+	
+  if ks component list | awk '{print $1}' | grep -q "^argo$"; then
+    ks param set argo workflowControllerImage ${registry}/workflow-controller:v2.2.0
+    ks param set argo uiImage ${registry}/argoui:v2.2.0
+    ks param set argo executorImage ${registry}/argoexec:v2.2.0
+  fi
+
+  if ks component list | awk '{print $1}' | grep -q "^ambassador$"; then
+    ks param set ambassador ambassadorImage ${registry}/datawire/ambassador:0.37.0
+  fi
+
+  if ks component list | awk '{print $1}' | grep -q "^katib$"; then
+    ks param set katib vizierDbImage ${registry}/mysql:8.0.3
+  fi
+
+  if ks component list | awk '{print $1}' | grep -q "^metacontroller$"; then
+    ks param set metacontroller image ${registry}/metacontroller:v0.3.0
+  fi
+}
+
```

Comment:
do you mind update pipeline's minio and mysql images 
https://github.com/kubeflow/kubeflow/blob/master/kubeflow/pipeline/pipeline.libsonnet#L9

nfs image can be ignored since it's not being used. 
Thanks!

Reply:
Done